frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +41,207 @@ export default function AddComments(props: Props) {
+                        event.preventDefault();
+                        disableHistoryScroll(false);
+                        setTarget(null);
+                        break;
+                    default:
+                        return;
+                }
+            }
+        },
+        [chars, editor, index, target]
+    );
+
+    const onKeyDown = (event: SyntheticKeyboardEvent<>) => {
+        if (event.key === 'Enter' && !event.shiftKey) {
+            event.preventDefault();
+            submitComment();
+        }
+    }
+    
     if (!user) {
         return null;
     }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
+    const onChange = (value) => {
+        setValue(value);
+        const { selection } = editor;
 
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
-
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
+        if (selection && Range.isCollapsed(selection)) {
+            const [start] = Range.edges(selection);
+            const wordBefore = Editor.before(editor, start, { unit: 'word' });
+            const before = wordBefore && Editor.before(editor, wordBefore);
+            const beforeRange = before && Editor.range(editor, before, start);
+            const beforeText = beforeRange && Editor.string(editor, beforeRange);
+            const beforeMatch = beforeText && beforeText.match(/^@(\w+)$/);
+            const after = Editor.after(editor, start);
+            const afterRange = Editor.range(editor, start, after);
+            const afterText = Editor.string(editor, afterRange);
+            const afterMatch = afterText.match(/^(\s|$)/);
+    
+            if (beforeMatch && afterMatch) {
+                setTarget(beforeRange);
+                setSearch(beforeMatch[1]);
+                setIndex(0);
+                return;
+            }
         }
-        else {
-            commentInput.current.rows = maxRows;
+    
+        setTarget(null);
+    }
+
+    const Portal = ({ children }) => {
+        if (!document.body) {
+            return null;
         }
+        return ReactDOM.createPortal(children, document.body);
     }
 
-    const handleOnKeyDown = (event: SyntheticKeyboardEvent<>) => {
-        if (event.keyCode === 13 && event.shiftKey === false) {
-            submitComment(event);
+    const disableHistoryScroll = (mentionsActive: boolean) => {
+        const element = document.querySelector(
+            '.history .mentions-disable-scroll'
+        );
+
+        if (!element) {
+            return null;
         }
+
+        mentionsActive 
+        ? element.style.overflow = 'hidden' 
+        : element.style.overflow = 'auto';
     }
 
-    const submitComment = (event: SyntheticEvent<>) => {
-        event.preventDefault();
-        const comment = commentInput.current.value;
+    const serialize = (node: any) => {
```

Comment:
I originally tried to define a type within 'types.js' but was unsuccessful. Given the way the data is handled I'm not quite sure how to tackle it. 

**Examples:**
With only a line of text this is what `node` logs:
<img width="421" alt="Screen Shot 2020-05-03 at 6 00 11 PM" src="https://user-images.githubusercontent.com/19564687/80930517-e6695780-8d68-11ea-8188-871773f54e68.png">

However, when there is a line of text and a mention this is what `node` logs:
<img width="730" alt="Screen Shot 2020-05-03 at 6 02 30 PM" src="https://user-images.githubusercontent.com/19564687/80930552-229cb800-8d69-11ea-9ea2-ab4f6ae40d63.png">

Reply:
You could define type like this:
```diff
-
-    const serialize = (node: any) => {
-        if (Text.isText(node)) {
+    type TextType = {|text: string|};
+    type MentionType = {|type: string, character: string, url: string, children: Array<TextType>|};
+    type InitialType = {|type: string, children: Array<TextType>|};
+    const serialize = (node: TextType | MentionType | InitialType) => {
+        if (Text.isText(node) && node.text) {
             return escapeHtml(node.text);
         }
 
+        if (!node.type || !node.children) {
+            return;
+        }
+
         const children = node.children.map(n => serialize(n)).join('');
 
         switch (node.type) {
             case 'paragraph':
                 return `<p>${children}</p>`;
             case 'link':
-                return `<a href="${escapeHtml(node.url)}">${children}</a>`;
+                if (node.url) {
+                    return `<a href="${escapeHtml(node.url)}">${children}</a>`;
+                }
             case 'mention':
-                return `<a href="${escapeHtml(node.url)}">${children}</a>`;
+                if (node.url) {
+                    return `<a href="${escapeHtml(node.url)}">${children}</a>`;
+                }
             default:
                 return children;
         }
```